torch/_inductor/ir.py

```diff
@@ -2808,8 +2808,9 @@ def require_stride1(cls, x):
 
     @classmethod
     def require_stride_order(cls, x, order):
-        if x.get_numel() == 0:  # Layout doesn't matter
-            return x
+        if not any([str(s) in V.graph.data_dep_symbol_to_expr_str for s in x.get_size()]):
```

Comment:
Actually, for data dependent symbols, we will typically (unsoundly) assume that they are nonzero, so conventional reasoning using SymInt value ranges would be able to infer if it's zero in the test here. In your AllToAllSingle, you probably failed to `constrain_range` appropriately; check nonzero for an example constrain range.